relayer/relayer.go

```diff
@@ -116,6 +124,51 @@ func NewRelayer(
 		)
 		return nil, nil, err
 	}
+
+	// Get the latest processed block height from the database.
+	var (
+		latestSeenBlockData []byte
+		latestSeenBlock     *big.Int // initialized to nil
+	)
+	latestSeenBlockData, err = r.db.Get(r.sourceChainID, []byte(database.LatestSeenBlockKey))
```

Comment:
Is this the latest seen block height, or the latest processed block height? We seem to use the two interchangeably in comments/variable names, but I think it makes a big difference in terms of guaranteeing we don't miss any messages because we may not have processed all messages in blocks that we have "seen".

Reply:
This is the latest **seen** block height, and should be across the entire application. I am not seeing any references to latest _processed_ block height, but please let me know if I missed any.

As far as behavior goes, we want to operate on the latest seen block height so that if the relayer exits mid-block, we can resume from that latest seen block, potentially re-processing warp messages, but not missing any. We store the latest seen block height before processing any of the warp messages contained therein.